docs/how-to-guides/block-tutorial/block-supports.md

```diff
@@ -0,0 +1,153 @@
+
+To avoid duplicating the same logic over and over in your blocks and to allign the behavior of your block with core blocks, you can make use of the different `supports` properties.
+
+Let's take the block we wrote in the previous chapter (example 3) and with just a single line of code, add support for text, link and background color customizations.
+
+Here's the exact same code we used to register the block previously.
+
+{% codetabs %}
+{% JSX %}
+
+```jsx
+import { registerBlockType } from '@wordpress/blocks';
+import { useBlockProps, RichText } from '@wordpress/block-editor';
+
+registerBlockType( 'gutenberg-examples/example-03-editable-esnext', {
+	apiVersion: 2,
+	title: 'Example: Basic with block supports',
+	icon: 'universal-access-alt',
+	category: 'design',
+	attributes: {
+		content: {
+			type: 'array',
+			source: 'children',
+			selector: 'p',
+		},
+	},
+	example: {
+		attributes: {
+			content: 'Hello World',
+		},
+	},
+	edit: ( props ) => {
+		const {
+			attributes: { content },
+			setAttributes,
+			className,
+		} = props;
+		const blockProps = useBlockProps();
+		const onChangeContent = ( newContent ) => {
+			setAttributes( { content: newContent } );
+		};
+		return (
+			<RichText
+				{ ...blockProps }
+				tagName="p"
+				onChange={ onChangeContent }
+				value={ content }
+			/>
+		);
+	},
+	save: ( props ) => {
+		const blockProps = useBlockProps.save();
+		return (
+			<RichText.Content
+				{ ...blockProps }
+				tagName="p"
+				value={ props.attributes.content }
+			/>
+		);
+	},
+} );
+```
+
+{% Plain %}
+
+```js
+( function ( blocks, blockEditor, element ) {
+	var el = element.createElement;
+	var RichText = blockEditor.RichText;
+	var useBlockProps = blockEditor.useBlockProps;
+
+	blocks.registerBlockType( 'gutenberg-examples/example-03-editable', {
+		apiVersion: 2,
+		title: 'Example: Basic with block supports',
+		icon: 'universal-access-alt',
+		category: 'design',
+
+		attributes: {
+			content: {
+				type: 'array',
+				source: 'children',
+				selector: 'p',
+			},
+		},
+		example: {
+			attributes: {
+				content: 'Hello World',
+			},
+		},
+		edit: function ( props ) {
+			var blockProps = useBlockProps();
+			var content = props.attributes.content;
+			function onChangeContent( newContent ) {
+				props.setAttributes( { content: newContent } );
+			}
+
+			return el(
+				RichText,
+				Object.assign( blockProps, {
+					tagName: 'p',
+					onChange: onChangeContent,
+					value: content,
+				} )
+			);
+		},
+
+		save: function ( props ) {
+			var blockProps = useBlockProps.save();
+			return el(
+				RichText.Content,
+				Object.assign( blockProps, {
+					tagName: 'p',
+					value: props.attributes.content,
+				} )
+			);
+		},
+	} );
+} )( window.wp.blocks, window.wp.blockEditor, window.wp.element );
+```
+
+{% end %}
+
+Now, let's alter the block.json file for that block, and add the supports key.
```

Comment:
Actually in the above example we don't use a `json` file to register the block, so it feels a bit unconnected here. I guess we should show the alternative by using `registerBlockType` with the args.

Reply:
block.json is not mandatory indeed, but we do show on the first step of the tutorial that we should start by block.json. 

I'm hesitant here, we can definitely just move the config to the `registerBlockType` function but kind of wanted to emphasize block.json as the best place for these configs. Maybe I can add a note to explain that we can also define these in `registerBlockType`

Reply:
A note sounds good 👍